src/expm.jl

```diff
@@ -69,47 +69,49 @@ end
 # Adapted from implementation in Base; algorithm from
 # Higham, "Functions of Matrices: Theory and Computation", SIAM, 2008
 function _exp(::Size, A::StaticMatrix{<:Any,<:Any,T}) where T
+    S = typeof((zero(T)*zero(T) + zero(T)*zero(T))/one(T))
+    A = S.(A)
```

Comment:
You'll need to rename this variable to make this fast on v0.6 (this inference "bug" has been fixed in v0.7, yay).